src/python/pants/engine/internals/graph_test.py

```diff
@@ -536,12 +530,40 @@ def test_dep_nocycle_indirect(transitive_targets_rule_runner: RuleRunner) -> Non
     }
 
 
-def test_resolve_generated_subtarget() -> None:
-    rule_runner = RuleRunner(target_types=[MockTarget])
-    rule_runner.write_files({"demo/BUILD": "target(sources=['f1.txt', 'f2.txt'])"})
-    generated_target_address = Address("demo", relative_file_path="f1.txt", target_name="demo")
-    generated_target = rule_runner.get_target(generated_target_address)
-    assert generated_target == MockTarget({Sources.alias: ["f1.txt"]}, generated_target_address)
+def test_resolve_generated_target(transitive_targets_rule_runner: RuleRunner) -> None:
```

Comment:
Most of this test is new, testing the rule of `Address -> WrappedTarget`.

Reply:
We still need more comprehensive tests for `UnexpandedTargets -> Targets`. All the functionality we care about is tested, but indirectly. I plan to fix that in a followup.